src/main/java/org/magicdgs/thaplv/tools/ibd/IBDcompute.java

```diff
@@ -0,0 +1,151 @@
+import org.magicdgs.thaplv.haplotypes.filters.NumberOfMissingFilter;
+import org.magicdgs.thaplv.io.FastaNsCounter;
+import org.magicdgs.thaplv.tools.ibd.engine.IBDOutput;
+import org.magicdgs.thaplv.tools.ibd.engine.IBDcollector;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFHeader;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.HaploidWalker;
+import org.broadinstitute.hellbender.engine.ReadsContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.filters.VariantFilter;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+/**
+ * Tool for computing IBD regions
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+@CommandLineProgramProperties(
+        summary =
+                "Identify identity-by-descent (IBD) regions using the approach described in Langley et. al 2012, Genetics 192(2). "
+                        + "Pairwise differences between individuals are computed in windows, and called as IBD tracks if the "
+                        + "differences per site are smaller than a threshold.",
+        oneLineSummary = "Compute IBD regions using pair-wise differences between individuals.",
+        programGroup = AlphaProgramGroup.class)
+public final class IBDcompute extends HaploidWalker {
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "Output prefix for the divergence (%s.diff) and IBD tracks (%s.ibd)", optional = true)
+    public String outputPrefix;
+
+    @Argument(fullName = ThaplvArgumentDefinitions.WINDOW_SIZE_LONG, shortName = ThaplvArgumentDefinitions.WINDOW_SIZE_SHORT, doc = "Window size for compute IBD regions (in bp)", optional = true)
+    public int windowSize = 500_000;
+
+    @Argument(fullName = ThaplvArgumentDefinitions.WINDOW_STEP_LONG, shortName = ThaplvArgumentDefinitions.WINDOW_STEP_SHORT, doc = "Step size for compute IBD regions (in bp)", optional = true)
+    public int stepSize = 100_000;
+
+    @Argument(fullName = "minimum-differences", shortName = "md", doc = "Threshold for number of differences per site in the window to output as an IBD", optional = true)
+    public double MINIMUM_DIFFERENCES = 0.0005;
+
+    @Argument(fullName = "output-differences", shortName = "out-diff", doc = "Output the divergence per window (.diff file) in addition to the IBD tracks")
+    public boolean outputDiff = false;
+
+    protected IBDOutput output;
+
+    private VCFHeader header;
+
+    @Override
+    public boolean requiresReference() {
```

Comment:
This requires the reference just for counting the number of Ns. Nevertheless, users could avoid this counting and consider the whole window. Add a TODO and open a ticket for include this behaviour.